101-automation-with-modules-and-runbook-execution/azuredeploy.json

```diff
@@ -0,0 +1,224 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "UserEmail": {
+      "type": "string",
+      "metadata": {
+        "description": "azuread email"
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "allowedValues": [
+        "japaneast",
+        "eastus2",
+        "westeurope",
+        "southeastasia",
+        "southcentralus",
+        "koreacentral",
+        "westus2",
+        "brazilsouth",
+        "uksouth",
+        "westcentralus",
+        "northeurope",
+        "canadacentral",
+        "australiasoutheast",
+        "centralindia"
+      ],
+      "metadata": {
+        "description": "location allowed for automation account."
+      }
+    },
+    "Password": {
+      "type": "securestring",
+      "metadata": {
+        "description": "aduser password"
+      }
+    },
+	 "WebAppName": {
+      "type": "string",
+      "metadata": {
+        "description": "This is the name of the web app that may be provided dynamically after creating."
+      }
+    },
+	 "WebAppResourceGroup": {
+      "type": "string",
+      "metadata": {
+        "description": "This is the name of the resource group in which web app resides."
+      }
+    },
+    "jobId": {
+      "type": "string",
+      "metadata": {
+          "description": "The GUID for the runbook job to be started."
+      }
+  },
+  "SubscriptionName":{
+    "type": "string",
+  "metadata": {
+      "description": "SubscriptionName"
+  }
+  }
+  },
+  "variables": {
+    "automationaccountname": "[concat('automation', uniquestring(resourceGroup().id))]",
+    "runbookName": "initialsetup",
+    "scriptUri": "https://storageformarketplace.file.core.windows.net/cblaze/1.0.0/initialSetup.ps1?sv=2017-11-09&ss=f&srt=sco&sp=r&se=3018-09-18T20:44:18Z&st=2018-09-18T12:44:18Z&spr=https,http&sig=1zNq%2FUzgU937NgY%2BmpqJaWXEfGqTncQ5UMdypkqxLPE%3D"
```

Comment:
See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#deployment-artifacts-nested-templates-scripts) for how to handle artifacts